packages/govuk-frontend/src/govuk/helpers/_colour.scss

```diff
@@ -10,30 +11,22 @@
 ///
 /// @param {String | Colour} $colour - Name of colour from the colour palette
 ///   (`$govuk-colours`)
-/// @param {String} $legacy - Either the name of colour from the legacy palette
-///   or a colour literal, to return instead when in 'legacy mode' - matching
-///   the palette from GOV.UK Template, Elements or Frontend Toolkit
 /// @return {Colour} Representation of named colour
 ///
-/// @example scss - Using legacy colours from the palette
-///  .foo {
-///    background-colour: govuk-colour("mid-grey", $legacy: "grey-2");
-///  }
-///
-/// @example scss - Using legacy colour literals
-///  .foo {
-///    background-colour: govuk-colour("green", $legacy: #BADA55);
-///  }
+/// @deprecated The `$legacy` parameter is deprecated and has been made
+///   non-operational, as the legacy colour palette has been removed.
+///   The parameter will be removed in the next major version.
 ///
 /// @throw if `$colour` is not a colour from the colour palette
 /// @access public
 
 @function govuk-colour($colour, $legacy: false) {
-  @if $govuk-use-legacy-palette and $legacy {
-    @if type-of($legacy) == "color" {
-      @return $legacy;
-    }
-    $colour: $legacy;
+  // Output a warning if $legacy is set to anything. This can't use the
+  // `_warning` mixin due to being a function, so we handle it manually.
+  @if $legacy and not index($govuk-suppressed-warnings, "legacy-palette") {
```

Comment:
**nitpick** `govuk-colour-legacy-param` may be more explicit, but I won't die on that hill 😊